.golangci.yml

```diff
@@ -54,6 +54,8 @@ linters:
   - whitespace
 
 linters-settings:
+  errchkjson:
+    check-error-free-encoding: true
```

Comment:
```suggestion
```

We should not set this to be equal to CAPI and resolve the findings instead by handling the error.

Reply:
Sure, I can make the necessary adjustments here, and it totally makes sense that we prefer to keep consistent with CAPI. 

To discuss it further:
This `check-error-free-encoding` parameter here, is just to indicate the `errchkjson` to ignore the errors that will never happen. 

Especially in some [cases:](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/9881385aeae28b7aea3d5e8aa5fedb0c48a46d33/packaging/flavorgen/flavors/generators.go#L131C2-L134C49)
```
sshKey, _ := json.Marshal(env.VSphereSSHAuthorizedKeysVar)
controlPlaneIP, _ := json.Marshal(env.ControlPlaneEndpointVar)
secretName, _ := json.Marshal(env.ClusterNameVar)
kubeVipPod, _ := json.Marshal(kubeVIPPodYaml())
```

the input of the `json.Marsha()` is just a string, which will never cause the function to fail, IIRC. To handle such a situation, is it a good idea to propagate errors to the higher-level calling function? Or do you recall any recommended approaches that I could follow?

Reply:
I did see that the required changes would only be in the flavorgen tool. So just bubbling them up should be fine.

Reply:
Got it. Then if I understand it correctly, all we need to do in this PR is just to enable `errchkjson` in `.golangci.yml` file. 😄

Reply:
Yes. No worries, just ask  all good :)

Reply:
Hm one moment. I mean when the linter can basically figure out for us that the error cannot happen, I think it makes sense to avoid the unnecessary noise from error handling.

Assuming I understood this correctly. I would prefer keeping this option and also introducing it in core CAPI. 

@chrischdi @killianmuldoon WDYT?

Reply:
also fine for me 👍

Reply:
Sorry @MrDXY for the back and forth :-)

Reply:
Alright. So keeping it as is is fine :)

Reply:
I'm fine with keeping it as it's the consensus - but I don't really like relying on external linters like this for something this fundamental, at the cost of saving a few lines of `if err == nil`'{}

Reply:
Good point on relying on an external linter. Basically if they have a bug in their logic we inherit that in CAPI as well

Reply:
Alright, changing my opinion again. Now I'm in favor of dropping the `check-error-free-encoding` config

Reply:
@chrischdi It's totally fine! Still, welcome any advice or suggestions you may have.
@sbueringer Got it!